app/search_cmd.go

```diff
@@ -16,6 +15,52 @@ type searchCmd struct {
 	JSONFormattable
 }
 
+// searchResult resolved from a manifest.
+type searchResult struct {
+	Name           string
+	Versions       []string
+	CurrentVersion string
+	Description    string
+}
+
+// buildSearchResult constructs a search result from packages with same name
+// p is an array expected to be package with same name
+func buildSearchResult(p []*manifest.Package) *searchResult {
+	out := &searchResult{
+		Versions: make([]string, 0),
+	}
+
+	if len(p) == 0 {
```

Comment:
Can this happen? Won't this result in an empty result in the output?

Reply:
just being cautious as I was doing `first = p[0]` below. making the set name & description into the loop now. which get rid of the need for this line.